Xamarin.Essentials/Permissions/Permissions.ios.tvos.macos.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Photos;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Permissions
+    {
+        public partial class Photos : BasePlatformPermission
+        {
+            protected override Func<IEnumerable<string>> RequiredInfoPlistKeys =>
+                () => new string[] { "NSPhotoLibraryUsageDescription" };
+
+            public override Task<PermissionStatus> CheckStatusAsync()
+            {
+                EnsureDeclared();
+
+                return Task.FromResult(GetPhotoPermissionStatus(PHAccessLevel.ReadWrite));
+            }
+
+            public override async Task<PermissionStatus> RequestAsync()
+            {
+                EnsureDeclared();
+
+                var status = GetPhotoPermissionStatus(PHAccessLevel.ReadWrite);
+                if (status == PermissionStatus.Granted)
+                {
+                    return status;
+                }
+#if __IOS__
+                else if (status == PermissionStatus.Limited)
+                {
+                    PhotosUI.PHPhotoLibrary_PhotosUISupport.PresentLimitedLibraryPicker(
+                        PHPhotoLibrary.SharedPhotoLibrary,
+                        Platform.GetCurrentUIViewController());
+                    return status;
+                }
+#endif
+
+                EnsureMainThread();
+
+                return await RequestPhotoPermissionStatus(PHAccessLevel.ReadWrite);
+            }
+        }
+
+        public partial class PhotosAddOnly : BasePlatformPermission
```

Comment:
Maybe we should call this just "AddPhotos" or "PhotosAdd" ?

Reply:
I think the current name is more understandable. But if you think it's too long, we can rename this class.